ci: Increase curl connect timeout to 15 seconds

gandro opened the pull request:
This increases the curl connection timeout from 5 to 15 seconds to avoid
issues with IPCache propagation delay. On Cilium master and 1.10, it
seems that IPCache updates in CI can take up to 4-8 seconds.

CI flakes potentially caused by the increased IPCache propagation delay:

 - #13839
 - #14959
 - #15103
 - #16237

gandro committed:
ci: Increase curl connect timeout to 15 seconds

This increases the curl connection timeout from 5 to 15 seconds to avoid
issues with IPCache propagation delay. On Cilium master an 1.10, it
seems that IPCache updates in CI can take up to 4-8 seconds.

CI flakes likely caused by the increased IPCache propagation delay:

 - #13839
 - #14959
 - #15103
 - #16237

Signed-off-by: Sebastian Wicki <sebastian@isovalent.com>

tklauser reviewed (approved):
Thanks!

nbusseneau commented on test/helpers/cons.go:
```diff
@@ -156,7 +156,9 @@ const (
 	// invokes
-	CurlConnectTimeout = 5
+	// FIXME: Revert this back to 5 seconds once the IPCache propagation delay
+	// (which seems to take up to 4-8 seconds) has been addressed.
```
[Nit] Can we add a link to a tracking issue?

gandro replied:
Done!

gandro committed:
ci: Make K8sCLI cilium bpf metrics more robust

This commit changes the `K8sCLI CLI Identity CLI testing Test cilium bpf
metrics list` test suite be more robust in terms of the expected "policy
deny" count, as it is dependent on the number of times a SYN packet is
retransmitted (e.g. a curl connection timeout of 5 creates usually 3 SYN
packets, but a timeout of 15 will generate 4 SYN packets). Instead of
hardcoding that number, we simply test that is is larger than zero.

Signed-off-by: Sebastian Wicki <sebastian@isovalent.com>
